### PR TITLE
Remove clap::ValueEnum from domain config enums (#320)

### LINKED ISSUE #320: Remove clap::ValueEnum from domain config enums (ColourPolicy, SpinnerMode, OutputFormat)
## Summary

`ColourPolicy`, `SpinnerMode`, and `OutputFormat` in `src/cli/config.rs` derive `clap::ValueEnum`, which couples the domain configuration schema to the Clap parsing layer. This makes the config types dependent on a CLI framework concern.

## Context

PR https://github.com/leynos/netsuke/pull/273 refactored these enums to use `clap::ValueEnum` as the mechanism for `FromStr` parsing (via `ValueEnum::from_str`). The `ParseEnumSpec` abstraction in `src/cli/parsing.rs` already centralises the Clap-facing dispatch, so the derive could be removed from the domain types by implementing a bespoke `FromStr` in `parsing.rs` instead.

## Proposed work

- Remove `#[derive(clap::ValueEnum)]` from `ColourPolicy`, `SpinnerMode`, and `OutputFormat` in `src/cli/config.rs`
- Implement the equivalent string-matching logic directly in the `parse_value_enum` helper in `src/cli/parsing.rs`
- Ensure `impl FromStr` for each type no longer depends on `ValueEnum::from_str`
- Verify all existing tests pass

## Backlinks

- PR: https://github.com/leynos/netsuke/pull/273
- Requested by @leynos

### PR DESCRIPTION
**Pull request by leynos**
## Summary

Closes #320

Decouples the domain configuration schema from the Clap parsing layer:

- `src/cli/config.rs`: drop `#[derive(clap::ValueEnum)]` from `ColourPolicy`, `SpinnerMode`, and `OutputFormat`; each `FromStr` now matches kebab-case names case-insensitively, mirroring the previous `ValueEnum::from_str(s, true)` behaviour exactly.
- `src/cli/parsing.rs`: `parse_value_enum` dispatches via `FromStr` instead of `ValueEnum`, keeping the Clap-facing dispatch centralised in `ParseEnumSpec` as proposed.
- `tests/bdd/steps/configuration_preferences.rs`: the steps for the three enums use `Display`/`str::parse` instead of `to_possible_value`/`ValueEnum::from_str`.

`Theme` keeps its derive — it parses through `ThemePreference::parse_raw` and is outside this issue's scope.

No behaviour change: clap's derived parser inference falls back to `FromStr`, and the runtime parsers were already the localised `parse_*` functions.

## Validation

- `make check-fmt` / `make lint` / `make test` — pass (37 suites, including all CLI parsing and configuration-preference BDD scenarios)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Decouple configuration enums from clap by parsing them via `FromStr` instead of `clap::ValueEnum` while preserving existing CLI behaviour.

Enhancements:
- Implement manual `FromStr` parsing for `ColourPolicy`, `SpinnerMode`, and `OutputFormat` enums with case-insensitive, kebab-case matching and remove their `ValueEnum` derives.
- Update generic CLI enum parsing to rely on `FromStr`-based parsing rather than `clap::ValueEnum` to reduce coupling to the argument parser.

Tests:
- Adjust configuration preference BDD steps to use enum `Display` and `FromStr`/`str::parse` instead of `ValueEnum` utilities, keeping test behaviour aligned with the new parsing model.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat o11y, code safety, documentation and validation coverage as in scope).

### Assessment against linked issues

| Issue | Objective | Addressed | Explanation |
| --- | --- | --- | --- |
| [#320](https://github.com/leynos/netsuke/issues/320) | Remove the `clap::ValueEnum` derive from `ColourPolicy`, `SpinnerMode`, and `OutputFormat` so the domain configuration enums are independent of Clap. | ❌ | The diff removes `ValueEnum` from `ColourPolicy` and several other policy enums, but it does not show changes to `SpinnerMode` or `OutputFormat`. Therefore, the issue's complete set of domain enums is not addressed. |

**Comment by buzzybee-df12** [comment]
@coderabbitai review

File: src/cli/parser.rs [hunk 1]
```diff
@@ -351,12 +317,26 @@ fn configure_validation_parsers(
...
-    let emoji_policy_parser = LocalizedValueParser::new(Arc::clone(localizer), parse_emoji_policy);
-    let progress_policy_parser =
-        LocalizedValueParser::new(Arc::clone(localizer), parse_progress_policy);
-    let accessibility_policy_parser =
-        LocalizedValueParser::new(Arc::clone(localizer), parse_accessibility_policy);
+    let color_policy_parser = LocalizedValueParser::with_possible_values(
+        Arc::clone(localizer),
+        parse_color_policy,
+        colour_policy_possible_values(),
+    );
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/parser.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/cli/value_parser.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+//! Clap (see [`super::policy_values`]).
+
+use clap::builder::PossibleValue;
+use clap::builder::TypedValueParser;
+use clap::error::ErrorKind;
+use ortho_config::Localizer;
+use std::sync::Arc;
+
+/// A Clap value parser that delegates validation to a localisation-aware
+/// closure and optionally advertises possible values for help rendering.
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/value_parser.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

~~~
## Individual Comments

### Comment 1
<location path="src/cli/parser.rs" line_range="320" />
<code_context>
-        LocalizedValueParser::new(Arc::clone(localizer), parse_progress_policy);
-    let accessibility_policy_parser =
-        LocalizedValueParser::new(Arc::clone(localizer), parse_accessibility_policy);
+    let color_policy_parser = LocalizedValueParser::with_possible_values(
+        Arc::clone(localizer),
+        parse_color_policy,
</code_context>
<issue_to_address>
**issue (broader_impact):** The new possible-value metadata is installed only by `configure_validation_parsers`, but the production `help` command renders `localize_command(Cli::command(), ...)` directly without calling that function. Consequently, `netsuke help` omits the policy flags' possible values and descriptions even though the parser help snapshot exercises the separately configured command.

**Triggers:** When users invoke the built-in `netsuke help` command instead of `--help`.

**Suggested fix:** Configure the validation parsers before rendering help in `render_root_help` (and any other production help path), or centralize command construction so both parsing and help use the same configured command.
</issue_to_address>

### Comment 2
<location path="src/cli/policy_values.rs" line_range="15" />
<code_context>
+/// descriptions carried by the `ColourPolicy` variants.
+pub fn colour_policy_possible_values() -> Vec<PossibleValue> {
+    vec![
+        PossibleValue::new("auto").help("Follow the host environment"),
+        PossibleValue::new("always").help("Force colour output on when available"),
+        PossibleValue::new("never").help("Force colour output off"),
</code_context>
<issue_to_address>
**issue (review_instructions):** Eliminate the duplicated policy metadata instead of hard-coding a second copy of each enum value and description.

These strings now have separate sources in `config.rs` and this module, so changing an enum variant or its documentation can silently leave CLI help stale. Centralize the possible-value metadata or derive it from one canonical definition.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*`

**Instructions:**
Keep code DRY, but readable. Use refactoring approaches best suited for the language in question.

[Edit this rule](https://app.sourcery.ai/accounts/147258/review-settings?activeTab=review-rules&rule=2f028dd9-3755-40cd-b8f9-b0829cfbac35&from_surface=github_bot&intent=review_rules)

</details>
</issue_to_address>
~~~

File: src/cli/value_parser.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#[derive(Clone)]
+pub(super) struct LocalizedValueParser<F> {
+    /// Shared localiser used to format validation messages.
+    localizer: Arc<dyn Localizer>,
+    /// Localiser-aware function validating each raw value.
+    parser: F,
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/value_parser.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/cli/value_parser.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/value_parser.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: src/cli/parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (leynos) [reply src/cli/parser.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat o11y, code safety, documentation and validation coverage as in scope).

### ❌ Failed checks (6 warnings)

| Check name | Status | Explanation | Resolution |
| :---: | :--- | :--- | :--- |
| Docstring Coverage | ⚠️ Warning | Docstring coverage is 64.00% which is insufficient. The required threshold is 80.00%. Docstring coverage is scoped to functions touched by this diff. Analyzed 25 functions across 10 files. | Write docstrings for the functions missing them to satisfy the coverage threshold. |
| User-Facing Documentation | ⚠️ Warning | Document the public API change. The PR removes the public `clap::ValueEnum` implementations from `ColourPolicy`, `ProgressPolicy`, `EmojiPolicy`, and `AccessibilityPolicy` in `src/cli/config.rs`. Thes… | Add a policy-parsing section to `docs/users-guide.md` that states the accepted values and case-insensitive CLI and configuration parsing. Add an entry to `docs/v0-1-0-migration-guide.md` that signposts removal of the public `clap::ValueEnum… |
| Developer Documentation | ⚠️ Warning | The pull request adds the internal `LocalizedValueParser` abstraction, its `with_possible_values` API, the `policy_values` adapter module, and a new configuration-to-Clap boundary. The diff contains n… | Update `docs/developers-guide.md` in the same change-set. Document the `src/cli` parsing boundary, the `LocalizedValueParser` contract and `with_possible_values` API, the Clap-only ownership of `policy_values`, and the build-script composit… |
| Testing (Unit And Behavioural) | ⚠️ Warning | Unit coverage is meaningful: `src/cli/config_tests.rs` checks lower-case, upper-case, mixed-case, and policy-specific errors for all four policy enums. However, the change also alters the public CLI p… | Add behavioural tests through `netsuke::cli::parse_with_localizer_from` or the BDD CLI boundary. Cover representative upper-case and mixed-case inputs for `--color`, `--emoji`, `--progress`, and `--accessibility`, and assert the resulting `… |
| Testing (Property / Proof) | ⚠️ Warning | Fail the check because the PR introduces a case-insensitive parsing invariant over a range of input spellings, but it adds only example-based `rstest` cases. `src/cli/config.rs` now lowercases input a… | Add Rust `proptest` tests for all four policy enums. Generate a canonical accepted value and independently vary the ASCII case of each letter, then assert that `str::parse` returns the expected variant. Retain the existing `rstest` checks f… |
| Testing (Compile-Time / Ui) | ⚠️ Warning | Add a dedicated compile-time/UI test for the changed Rust trait surface. The PR removes `clap::ValueEnum` from public `ColourPolicy`, `ProgressPolicy`, `EmojiPolicy`, and `AccessibilityPolicy`, and ch… | Add and register a trybuild test, or an equivalent direct-rustc UI test, for the public policy API. Cover successful external use through `FromStr` and the intended compile-time `ValueEnum` decoupling. Keep the existing focused help snapsho… |

**Comment by buzzybee-df12** [comment]
@coderabbitai review
